src/Features/Core/Portable/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -60,7 +61,11 @@ private static TextSpan GetLineSpanForReference(SourceText sourceText, TextSpan
             var sourceLine = sourceText.Lines.GetLineFromPosition(referenceSpan.Start);
             var firstNonWhitespacePosition = sourceLine.GetFirstNonWhitespacePosition().Value;
 
-            return TextSpan.FromBounds(firstNonWhitespacePosition, sourceLine.End);
+            // Get the span of the line from the first non-whitespace character to the end of it. Note: the reference
+            // span might actually start in the leading whitespace of the line (nothing prevents any of our
+            // languages/providers from doing that), so ensure that the line snap we clip out at least starts at that
```

Comment:
Does this indicate a bug though that we should be tracking down as well?

Reply:
For FAR (and for C#/VB), very likely yes.  THough we're at a part where we don't what feature/purpose this code is being called for.  For example, it is also called during async go-to-def, where the features generate a result that then generates these UI elements based on a DocId+TextSpan.  So in that codepath, it's very possible for an edit to then affect the buffer versus what the snapshot result said.  

To fix this, which we likely should, we probably need a way to update IDocumentNavigationService.GetLocationForSpanAsync to actually take a Document instance, not a DocId.  I've looked into this in the past, and it's a substantial refactoring, and has major concerns about accidentally holding solution-snapshot objects around too long.  So, for now, this is just in the realm of keeping things working the same way, and keeping hte code resilient to data like this coming through.